src/site/stages/build/plugins/inject-axe-core.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-continue, no-param-reassign */
+
+require('axe-core');
+
+const path = require('path');
+const ENVIRONMENTS = require('../../../constants/environments');
+
+const axeSource = module.children.find(
+  el => el.filename.indexOf('axe-core') !== -1,
+).exports.source;
+
+function executeAxeCheck() {
+  // eslint-disable-next-line no-undef
+  axe.run(
+    'main',
+    {
+      iframes: false,
+      runOnly: {
+        type: 'tag',
+        values: ['section508', 'wcag2a', 'wcag2aa', 'best-practice'],
+        resultTypes: ['violations'],
+      },
+      rules: {
+        'color-contrast': { enabled: false },
+      },
+    },
+    (error, results) => {
+      if (results.violations.length > 0) {
+        const bannerEl = document.createElement('div');
+
+        bannerEl.innerHTML = `
+          <details class="vads-u-background-color--gold-lighter vads-u-padding--1">
+            <summary>There are (${
+              results.violations.length
+            }) accessibility issues on this page.</summary>
+            <ul class="usa-unstyled-list vads-u-padding-y--2 vads-u-padding-x--6">
+              ${results.violations
+                .map(violation => {
+                  const {
+                    description,
+                    help,
+                    helpUrl,
+                    impact,
+                    nodes,
+                    tags,
+                  } = violation;
+
+                  return `
+                  <li class="vads-u-margin-y--1">
+                    <details>
+                      <summary>${help}</summary>
+                      <ul class="usa-unstyled-list vads-u-padding-y--1 vads-u-padding-x--2">
+                        <li><strong>Description</strong>: ${description}</li>
+                        <li><strong>Impact</strong>: ${impact}</li>
+                        <li><strong>Tags</strong>: ${tags.join(', ')}</li>
+                        <li><strong>Help</strong>: <a href="${helpUrl}" target="blank" rel="noopener noreferrer">${helpUrl}</a></li>
+                        <li><strong>HTML</strong>:
+                          <ol>
+                            <li>
+                              ${nodes
+                                .map(node => {
+                                  const code = document.createElement('code');
+                                  code.innerText = node.html;
+                                  return code.outerHTML;
+                                })
+                                .join('')}
+                            </li>
+                          </ol>
+                        </li>
+                      </ul>
+                    </details>
+                  </li>
+                `;
+                })
+                .join('')}
+            </ul>
+          </details>
+        </div>
+      `;
+
+        const header = document.querySelector('header');
+
+        header.prepend(bannerEl, header.firstChild);
```

Comment:
This code is ES6 but it's injected directly into the browser, so it isn't compiled down to ES5. So, we either need to convert it to ES5 or move this whole render into the Webpack build, probably as a React component. Or, consider it okay as-is since it isn't user-facing (although CMS users do see it. I wonder if some of them are using IE 🤔 ) I really don't want to move it into React because I like this build step being self-contained.

Reply:
@ncksllvn This is magical. I'll give it a look tonight after dinner. I'm helping Rian with another IE-specific issue.

I think it'll be worth stepping down to ES5 because there's at least a chance IE11 is still being used for the moment. I'm hopeful Edge will be released on Chromium soon and adopted.

Reply:
Oof, just realized the `≤details>` element isn't supported either by IE 🤔